subsystems/networking/portal_linux.go

```diff
@@ -176,7 +176,7 @@ func (n *Networking) portalSave(resp http.ResponseWriter, req *http.Request) {
 			n.errors.Add(errw.Wrap(err, "invalid json config contents"))
 			return
 		}
-		if cfg.Cloud == nil || (cfg.Cloud.ID == "" || cfg.Cloud.Secret == "" || cfg.Cloud.AppAddress == "") {
+		if cfg.Cloud == nil || (cfg.Cloud.ID == "" || (cfg.Cloud.Secret == "" || cfg.Cloud.APIKey.IsEmpty()) || cfg.Cloud.AppAddress == "") {
```

Comment:
Not sure what this function does and I do see that we still have `n.portalData.input.Secret = cloud.GetSecret()` in the other file, but do we indeed want to require that both a secret and API key is present here?

Mostly wondering about the intention since wrapping the two in parentheses is redundant / it goes against how you tested connecting.